platform/server-config/src/main/java/org/motechproject/config/service/impl/ConfigurationServiceImpl.java

```diff
@@ -647,7 +648,7 @@ public void removeBundleRecords(List<ModulePropertiesRecord> records) {
      * @param settingsRecord  the settings to be add
      */
     @Transactional
-    public void addOrUpdateSettings(SettingsRecord settingsRecord) {
+    private void addOrUpdateSettings(SettingsRecord settingsRecord) {
```

Comment:
According to spring docs (http://docs.spring.io/spring/docs/3.0.x/spring-framework-reference/html/transaction.html#transaction-declarative-annotations) :

When using proxies, you should apply the @Transactional annotation only to methods with public visibility. If you do annotate protected, private or package-visible methods with the @Transactional annotation, no error is raised, but the annotated method does not exhibit the configured transactional settings. Consider the use of AspectJ (see below) if you need to annotate non-public methods.

In this method we need transaction, so it should be public.